### PR TITLE
Add new RemoteApplicationInfo api

### PR DESCRIPTION
**Pull request by wallyworld**
A new api is added to the Application facade:
RemoteApplicationInfo()

This will be used by the GUI to get metadata about remote applications that appear on the canvas.

QA: smoke test only, no callers of the api yet

### CODE REVIEW

File: state/allwatcher.go [hunk 1]
```diff
@@ -401,24 +401,24 @@ func (u *backingUnit) mongoId() string {
 
 type backingApplication applicationDoc
 
-func (svc *backingApplication) updated(st *State, store *multiwatcherStore, id string) error {
-	if svc.CharmURL == nil {
+func (app *backingApplication) updated(st *State, store *multiwatcherStore, id string) error {
```
Comment (babbageclunk) [inline state/allwatcher.go]: nice driveby

**Comment by wallyworld** [comment]
$$merge$$
